packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -760,7 +776,18 @@ class ScaffoldState extends State<Scaffold> {
     } else {
       children.add(new LayoutId(child: _buildScrollableAppBar(context, padding), id: _ScaffoldSlot.appBar));
     }
-    // Otherwise the AppBar will be part of a [app bar, body] Stack. See AppBarBehavior.scroll below.
+    // Otherwise the AppBar will be part of a [app bar, body] Stack. See
+    // AppBarBehavior.scroll below.
+
+    if (_snackBars.isNotEmpty)
+      _addIfNonNull(children, _snackBars.first._widget, _ScaffoldSlot.snackBar);
```

Comment:
Well, according to the spec, snack bars should be below bottom navigation and bottom sheets should be above bottom navigation. There wasn't any rule for snack bars and bottom sheets, but you can infer the order now that bottom navigation is in the equation.